src/wire/ipv6hop.rs

```diff
@@ -0,0 +1,307 @@
+use core::fmt;
+use {Error, Result};
+
+pub use super::IpProtocol as Protocol;
+
+/// A read/write wrapper around an IPv6 Hop-by-Hop Options Header.
+#[derive(Debug, PartialEq)]
+pub struct Packet<T: AsRef<[u8]>> {
+    buffer: T
+}
+
+// Format of the Hop-by-Hop Options Header
+//
+// +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+// |  Next Header  |  Hdr Ext Len  |                               |
+// +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+                               +
+// |                                                               |
+// .                                                               .
+// .                            Options                            .
+// .                                                               .
+// |                                                               |
+// +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+//
+//
+// See https://tools.ietf.org/html/rfc8200#section-4.3 for details.
+mod field {
+    #![allow(non_snake_case)]
+
+    use wire::field::*;
+
+    // Minimum size of the header.
+    pub const MIN_HEADER_SIZE: usize = 8;
+
+    // 8-bit identifier of the header immediately following this header.
+    pub const NXT_HDR:  usize = 0;
+    // 8-bit unsigned integer. Length of the OPTIONS field in 8-octet units,
+    // not including the first 8 octets.
+    pub const LENGTH:   usize = 1;
+    // Variable-length field. Option-Type-specific data.
```

Comment:
There's no Option Type here, is there?

Reply:
No, there may be multiple options present.